R/dependencies.R

```diff
@@ -0,0 +1,112 @@
+##' @title Get the dependencies for a given report from the orderly_run.yml
```

Comment:
I am confused about the two `get_dependencies` and `what_depends_on` - as names it's not clear which I should reach for when as they seem very similar